File: 0682/sesión25 - Reportes/Back-End/ServicioRESTFul/SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/ReportesStocksDTO.java
package pe.edu.pucp.softinv.model;

public class ReportesStocksDTO {

    private PeriodosDTO periodo;
    private ProductosDTO producto;
    private AlmacenesDTO almacen;
    private Double saldoInicial;
    private Integer entradas;
    private Integer salidas;
    private Double saldoFinal;

    public ReportesStocksDTO() {
        this.periodo = null;
        this.producto = null;
        this.almacen = null;
        this.saldoInicial = null;
        this.entradas = null;
        this.salidas = null;
        this.saldoFinal = null;
    }

    public ReportesStocksDTO(PeriodosDTO periodo, ProductosDTO producto, AlmacenesDTO almacen, Double saldoInicial, Integer entradas, Integer salidas, Double saldoFinal) {
        this.periodo = periodo;
        this.producto = producto;
        this.almacen = almacen;
        this.saldoInicial = saldoInicial;
        this.entradas = entradas;
        this.salidas = salidas;
        this.saldoFinal = saldoFinal;
    }

    /**
     * @return the periodo
     */
    public PeriodosDTO getPeriodo() {
        return periodo;
    }

    /**
     * @param periodo the periodo to set
     */
    public void setPeriodo(PeriodosDTO periodo) {
        this.periodo = periodo;
    }

    /**
     * @return the producto
     */
    public ProductosDTO getProducto() {
        return producto;
    }

    /**
     * @param producto the producto to set
     */
    public void setProducto(ProductosDTO producto) {
        this.producto = producto;
    }

    /**
     * @return the almacen
     */
    public AlmacenesDTO getAlmacen() {
        return almacen;
    }

    /**
     * @param almacen the almacen to set
     */
    public void setAlmacen(AlmacenesDTO almacen) {
        this.almacen = almacen;
    }

    /**
     * @return the saldoInicial
     */
    public Double getSaldoInicial() {
        return saldoInicial;
    }

    /**
     * @param saldoInicial the saldoInicial to set
     */
    public void setSaldoInicial(Double saldoInicial) {
        this.saldoInicial = saldoInicial;
    }

    /**
     * @return the entradas
     */
    public Integer getEntradas() {
        return entradas;
    }

    /**
     * @param entradas the entradas to set
     */
    public void setEntradas(Integer entradas) {
        this.entradas = entradas;
    }

    /**
     * @return the salidas
     */
    public Integer getSalidas() {
        return salidas;
    }

    /**
     * @param salidas the salidas to set
     */
    public void setSalidas(Integer salidas) {
        this.salidas = salidas;
    }

    /**
     * @return the saldoFinal
     */
    public Double getSaldoFinal() {
        return saldoFinal;
    }

    /**
     * @param saldoFinal the saldoFinal to set
     */
    public void setSaldoFinal(Double saldoFinal) {
        this.saldoFinal = saldoFinal;
    }
}

File: 0683/sesión 15  - Master Detail/sin Master Detail/SoftInv/SoftInvModel/ReportesStocksDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftInv.Model
{
    public class ReportesStocksDTO
    {
        private PeriodosDTO periodo;
        private ProductosDTO producto;
        private AlmacenesDTO almacen;
        private int? saldoInicial;
        private int? entradas;
        private int? salidas;
        private int? saldoFinal;

        public ReportesStocksDTO()
        {
            this.Periodo = null;
            this.Producto = null;
            this.Almacen = null;
            this.SaldoInicial = null;
            this.Entradas = null;
            this.Salidas = null;
            this.SaldoFinal = null;
        }

        public ReportesStocksDTO(PeriodosDTO periodo, ProductosDTO producto, AlmacenesDTO almacen, int saldoInicial, int entradas, int salidas, int saldoFinal)
        {
            this.Periodo = periodo;
            this.Producto = producto;
            this.Almacen = almacen;
            this.SaldoInicial = saldoInicial;
            this.Entradas = entradas;
            this.Salidas = salidas;
            this.SaldoFinal = saldoFinal;
        }

        public PeriodosDTO Periodo { get => periodo; set => periodo = value; }
        public ProductosDTO Producto { get => producto; set => producto = value; }
        public AlmacenesDTO Almacen { get => almacen; set => almacen = value; }
        public int? SaldoInicial { get => saldoInicial; set => saldoInicial = value; }
        public int? Entradas { get => entradas; set => entradas = value; }
        public int? Salidas { get => salidas; set => salidas = value; }
        public int? SaldoFinal { get => saldoFinal; set => saldoFinal = value; }
    }
}
